src/exec.h

```diff
@@ -3,6 +3,8 @@
 
 #include <vector>
 
-void exec(const std::vector<unsigned char> &obj, int argc, char *argv[]);
+#include "debuginfo.h"
```

Comment:
Use a forward declaration here (`class DebugInfo;`) instead of including the whole header when it's not needed.